src/LocationSelector.vue

```diff
@@ -447,15 +462,39 @@ export default defineComponent({
       return [location.latitudeDeg, location.longitudeDeg];
     },
 
+    updateRectangleIntensity(): void {
+      (this.cloudCoverRectangles as L.LayerGroup<L.Rectangle>).eachLayer((layer) => {
+        if (layer instanceof L.Rectangle) {
+          const latLng = layer.getBounds().getCenter();
+          const lat = latLng.lat;
+          const lon = latLng.lng;
+          const key = `${lat},${lon}`;
+
+          const cloudCover = this.selectedCloudCover[this.index[key]].cloudCover;
+          if (cloudCover !== undefined) {
+            layer.setStyle({fillOpacity: this.cloudCoverOpacityFunction(cloudCover), opacity: cloudCover });
+          }
+        }
+      });
+    },
+
+
     updateCloudCover(value: boolean) {
-      if (value && this.selectedCloudCover != null) {
-        this.cloudCoverRectangles.remove(); // do we need to remove it from the map? but how do we make sure it's not already on the map?
-        this.cloudCoverRectangles.clearLayers(); // clear the rectangles is what we want
-        this.parseResult(this.selectedCloudCover);
+      if (value) {
+        // If rectangles have been created already, update their intensity
+        if (this.rectanglesCreated) {
+          this.updateRectangleIntensity();
+        } else {
+          // If rectangles haven't been created yet, run parseResult to create them
+          this.parseResult(this.selectedCloudCover);
+          this.rectanglesCreated = true; // Set the flag to true
+        }
       } else {
-        this.cloudCoverRectangles.remove();
+        // Clear cloud cover rectangles if value is false
+        this.cloudCoverRectangles.clearLayers();
```

Comment:
If the user's hiding the cloud cover, would it be better to just set all of the opacities to zero, rather than delete the rectangles (and then have to re-create)? This update is a performance optimization, and based on the results so far I would imagine that'd make both hiding and a potential future re-display be faster.